File: suites/02_numerical_computation/comparisons/polynomial_eval.c
#include <stdio.h>
static long long polynomial_cost(long long points, long long degree) {
    long long checksum = 0;
    for (long long p = 0; p < points; ++p) {
        long long x = (p % 97) + 3, value = 1;
        for (long long c = degree; c > 0; --c)
            value = ((value * x) + ((c * 11) + (p % 29))) % 1000003;
        checksum += value;
    }
    return checksum;
}
int main(void) { printf("%lld\n", polynomial_cost(800000, 16)); return 0; }

File: suites/02_numerical_computation/comparisons/polynomial_eval.rs
fn polynomial_cost(points: i64, degree: i64) -> i64 {
    let mut checksum: i64 = 0;
    for p in 0..points {
        let x = (p % 97) + 3;
        let mut value: i64 = 1;
        let mut c = degree;
        while c > 0 { value = ((value * x) + ((c * 11) + (p % 29))) % 1000003; c -= 1; }
        checksum += value;
    }
    checksum
}
fn main() { println!("{}", polynomial_cost(800000, 16)); }
